File: app/src/main/java/com/lucas/omnia/models/Post.java
package com.lucas.omnia.models;

/**
 * Created by Lucas on 24/10/2017.
 */
import com.google.firebase.database.Exclude;
import com.google.firebase.database.IgnoreExtraProperties;

import java.util.HashMap;
import java.util.Map;

@IgnoreExtraProperties
public class Post {

    public String uid;
    public String author;
    public long timestamp;
    public String title;
    public String body;
    public boolean edited;
    public boolean hasImage;
    public int upVoteCount = 0;
    public int downVoteCount = 0;
    public int commentCount = 0;
    public Map<String, Boolean> upVotes = new HashMap<>();
    public Map<String, Boolean> downVotes = new HashMap<>();

    public Post() {
        // Default constructor required for calls to DataSnapshot.getValue(Post.class)
    }

    public Post(String uid, String author, String title, String body) {
        this.uid = uid;
        this.author = author;
        this.title = title;
        this.body = body;
    }

    @Exclude
    public Map<String, Object> toMap() {
        HashMap<String, Object> result = new HashMap<>();
        result.put("uid", uid);
        result.put("author", author);
        result.put("timestamp", timestamp);
        result.put("title", title);
        result.put("body", body);
        result.put("upVoteCount", upVoteCount);
        result.put("downVoteCount", downVoteCount);
        result.put("commentCount", commentCount);

        return result;
    }
}

File: app/src/main/java/com/lucas/omnia/models/Post.kt
package com.lucas.omnia.models

import com.google.firebase.database.Exclude
import com.google.firebase.database.IgnoreExtraProperties
import java.util.*

@IgnoreExtraProperties
data class Post(var uid: String? = null, var author: String? = null, var title: String? = null, var body: String? = null) {
    @JvmField
    var timestamp: Long = 0
    @JvmField
    var edited = false
    @JvmField
    var hasImage = false
    @JvmField
    var upVoteCount = 0
    @JvmField
    var downVoteCount = 0
    @JvmField
    var votesBalance = 0
    @JvmField
    var commentCount = 0
    @JvmField
    var upVotes: MutableMap<String, Boolean> = HashMap()
    @JvmField
    var downVotes: MutableMap<String, Boolean> = HashMap()

    // Default constructor required for calls to DataSnapshot.getValue(Post.class)
    constructor() : this("", "", "", "")

    @Exclude
    fun toMap(): Map<String, Any?> {
        val result = HashMap<String, Any?>()
        result["uid"] = uid
        result["author"] = author
        result["timestamp"] = timestamp
        result["title"] = title
        result["body"] = body
        result["upVoteCount"] = upVoteCount
        result["downVoteCount"] = downVoteCount
        result["votesBalance"] = votesBalance
        result["commentCount"] = commentCount
        return result
    }
}